### DISCUSSION TITLE
Can OpenDDS communicate with RTI?

### DISCUSSION THREAD

**Discussion by amielc1** [OpenDDS/OpenDDS#4667] [Q&A]
We use Opendds, but some consumers will use RTI - does it work? 
1. when we use via RtpsRelay
2. with the regular use - multicast 

Thank a lot

**Comment by jrw972** [comment]
For (2) OpenDDS and RTI are interoperable when using multicast/unicast as defined by the RTPS spec.  Explore https://github.com/omg-dds/dds-rtps/.

For (1), as far as I know, RTI has not extended their product to use the RtpsRelay.
A possible way of working around this would be to create a lightweight bridge between RTI and the RtpsRelay.
Such a bridge would need to

1. Execute the STUN protocol to keep NAT port bindings alive
2. Forward SPDP, SEDP, and the user data messages to the RtpsRelay.

The cloud-multicast-repeater might be a good starting point for such a utility.

On the other hand, if you can configure everything to use unicast, no firewalls are in the way, etc. so that the only issue is SPDP multicast, then extending the RtpsRelay with a UDP version of https://github.com/OpenDDS/OpenDDS/wiki/Proposal:-RtpsRelay-Improvements#tcp-based-spdp-repeater-tsr-for-opendds-4 would probably be the way to go.
Such a solution would assume that you can force an RTI participant to send SPDP to an arbitrary IP address.
